src/lease-process/lease-process-pool.ts

```diff
@@ -322,6 +312,12 @@ export class LeaseProcessPool {
           ),
         ),
       );
+      // Wait for at least 1 tick before trying again
+      // otherwise there is a risk of blocking the event loop and:
+      // a) the pool will never get the chance to gain more offers
+      // b) the abort signal will never change, because timers are processed once a tick
+      // leading to an infinite loop
+      await sleep(0);
```

Comment:
That's suuuuper smelly. Points to a bad decision somewhere else. Where's the root cause?

Reply:
This is the infinite loop I was referring to yesterday, if `createNewLeaseProcess` rejects or resolves immediately this will loop infinitely (most noticeable when mocked in tests). Using a `setTImeout` to wait for a couple ticks is probably the cleanest way to fix this (`setImmediete` is not available in browsers and using `requestAnimationFrame` here feels just wrong)